### PR TITLE
Mute ads automatically

### LINKED ISSUE #112: Adblocking
I might try to tackle this myself in a while

### PR DESCRIPTION
**Pull request by vincens2005**
closes #112

### CODE REVIEW

File: src/preload.js [hunk 1]
```diff
@@ -42,7 +43,7 @@ const elements = {
   playing_title: 'span[data-test="table-cell-title"].css-geqnfr',
   album_name_cell: '[data-test="table-cell-album"]',
   tracklist_row: '[data-test="tracklist-row"]',
-
+  mute: '*[data-test="volume"]',
```
Comment (Mastermindzh) [inline src/preload.js:46]: This should probably be 'volume'.

File: src/preload.js [hunk 2]
```diff
@@ -350,6 +355,15 @@ setInterval(function () {
   const progressBarTimeChanged = progressBarcurrentTime !== progressBarTime;
   const titleOrArtistChanged = currentSong !== songDashArtistTitle;
 
+
+  if (store.get(settings.muteAds)) {
+    if (artists == "TIDAL") {
```
Comment (Mastermindzh) [inline src/preload.js:360]: Can be a triple `=` 😄

File: src/preload.js [hunk 3]
```diff
@@ -350,6 +355,15 @@ setInterval(function () {
...
+      isAd = true;
+      if (!elements.isMuted()) elements.click("mute");
```
Comment (Mastermindzh) [inline src/preload.js:362]: Please add brackets, keeping the code consistent is quite important

**Review by Mastermindzh — APPROVED** [review]
I have a few small remarks and... 1 big one.

I am **unsure** whether I want to deal with any ad-blocking in Tidal-Hifi.
It being an option is... good. But I need the weekend to think about this.

Right now I'm thinking it isn't a bad idea to have it since we're not blocking anything but simply muting it. 

I'll let you know, thanks for the contribution anyway!

**Comment by vincens2005** [comment]
Okay, I've made the changes you suggested. I understand if you don't want this functionality in the app, since it goes against what Tidal themselves would want.

**Comment by Mastermindzh** [comment]
@vincens2005, indeed. But I've thought of something more generic that can also do this.

We can add a setting (idc if it's a comma-separated list or something fancy) to mute "certain artists".
This PR is a great basis for that. (We can add `Tidal` as the default)

Let me know if you want to take this on, if not I will do it later this week and merge it 😄 

Thanks for the contribution!

**Comment by vincens2005** [comment]
That's a good idea. I don't have a lot of time this week, but I can do it later on if you'd like

**Pull request merged**